mysql-ballerina/build.gradle

```diff
@@ -278,22 +278,75 @@ publishing {
     }
 }
 
+def checkExecResult(execResult, failText, standardOutput){
+    if (execResult) {
+        if (execResult.getExitValue() != 0) {
+            throw new GradleException('Non-zero exit value: ' + execResult.getExitValue())
+        }
+        if (standardOutput.toString().contains(failText)){
+            throw new GradleException('"' + failText + '" string in output: ' + standardOutput.toString())
+        }
+    } else {
+        throw new GradleException('Returned a null execResult object')
+    }
+}
+
+task createTestDockerImage(type: Exec) {
+    if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+        def standardOutput = new ByteArrayOutputStream()
+        commandLine 'sh', '-c', "docker build -f $project.projectDir/tests/resources/Dockerfile -t ballerina-mysql" +
+                " -q $project.projectDir/tests/resources/"
+        doLast {
+            checkExecResult(execResult, 'Error', standardOutput)
+            sleep(10 * 1000)
+        }
+    }
+}
+
+
+task startTestDockerContainer(type: Exec) {
+     if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+         def standardOutput = new ByteArrayOutputStream()
+         commandLine 'sh', '-c', "docker run --rm -d --name ballerina-mysql -p 3305:3306 -t ballerina-mysql"
+         doLast {
+             checkExecResult(execResult, 'Error', standardOutput)
+             sleep(20 * 1000)
+         }
+    }
+}
+
+task stopTestDockerContainer() {
+    doLast {
+        if (!Os.isFamily(Os.FAMILY_WINDOWS)) {
+            def stdOut = new ByteArrayOutputStream()
+            exec {
+                commandLine 'sh', '-c', "docker stop ballerina-mysql"
+                standardOutput = stdOut
+            }
+        }
+    }
+}
+
 unpackJballerinaTools.dependsOn copyToLib
 unpackStdLibs.dependsOn unpackJballerinaTools
 copyStdlibs.dependsOn unpackStdLibs
 updateTomlFile.dependsOn copyStdlibs
+startTestDockerContainer.dependsOn createTestDockerImage
 
 ballerinaTest.dependsOn updateTomlFile
 ballerinaTest.dependsOn initializeVariables
 ballerinaTest.dependsOn ":${packageName}-native:build"
+ballerinaTest.dependsOn startTestDockerContainer
 ballerinaTest.finalizedBy revertTomlFile
+ballerinaTest.finalizedBy stopTestDockerContainer
 test.dependsOn ballerinaTest
 
 ballerinaBuild.dependsOn test
 ballerinaBuild.dependsOn updateTomlFile
 ballerinaBuild.dependsOn initializeVariables
 ballerinaBuild.dependsOn ":${packageName}-native:build"
 ballerinaBuild.finalizedBy revertTomlFile
+ballerinaBuild.finalizedBy stopTestDockerContainer
```

Comment:
Is this necessary? If the container is not started before build? Only needed for test isnt it?

Reply:
Yes, because when "./gradlew clean build"  command issued,  the stopTestDockerContainer task runs before running the test cases which leads to a failure in testing.